doc/example/example_nonlinear_monotone.ipynb

```diff
@@ -0,0 +1,856 @@
RwAAACwsMpItrp166aTJ0/Kzc2t3OdMmjRJiYmJlRrXzRw+fFgLFy7UjBkzdPz4cbO2nj17ymAwmL2ef/55sz7Hjx9X//795eTkpAYNGugvf/mLrly5YtbnxIkTWrBggd555x3t2bPH4vfwxhtvyNbWVgsWLKjQeceOHZPBYFBqaqrZ8cWLFys2Nrbc49SE7xEAAAAArBU5/B9qYw5/7tw5eXt7a+7cuSXaQkJC1KVLFxUUFNz2dWAdKIwDAAAAtYCDg4O8vb1lMBjKfY6Li4s8PT0tHkteXl65+/7666+SpOHDh6tx48Yl2seNG6eTJ0+aXvPnzze1FRQUqH///srLy9PWrVsVFxen2NhYvf7662ZjNG7cWGFhYZKkX3755VZu6YZiYmI0efJkxcTEWGQ8Nze3mz4pcK3K+h4BAAAAAJWDHL7qcvj69etr6dKlmjlzpvbt22c6vnr1aq1fv15xcXGV9vS7JOXn51fa2LA8CuMAAAC4o/yecb7Kr9mzZ09NnDhRkydPVr169eTt7a3IyMgS/c6dO6fBgwfLyclJfn5++uyzz0xt1y/DVmzdunXy8/OT0WhUcHCwTpw4YWq7dhm2yMhIxcXF6dNPPzXN7E5KSpIk7du3T0FBQXJ0dJSnp6eee+45Xbx40TRO8Sz1OXPmyMfHR61bty73vRcWFkqS7OzsSm13cnKSt7e36eXq6mpq++qrr3Tw4EF98MEHat++vfr27avZs2fr7bffLpHY29vbS5LFZ4F/++23unz5smbNmqWsrCxt3brVrL2wsFDz58+Xr6+v6tSpo7vuuktz5syRJLVo0UKS1KFDBxkMBvXs2VOS+az/pUuXysfHx/Q5FXv88cf17LPPSirf9xgUFKQJEyaYjXH27Fk5ODiYPXEAAAAAALUJOTw5fHkMGjRII0eO1OjRo5Wfn6+zZ89q/PjxmjdvXon7nzlzpry8vOTq6qrnn3/eLLYNGzbowQcflLu7uzw9PTVgwAAdOXLE1F68MtyqVav08MMPy2g0KiEhwSL3gKpBYRwAAAC1Tn5OToVeexM3mM6NfeV57d284WpbXm65xrWEuLg4OTs7a8eOHZo/f75mzZqlTZs2mfWZOXOmQkJCtHfvXvXr109hYWHKyMgoc8xLly5pzpw5io+PV3JysjIzMzVixIhS+06aNEkhISHq06ePaWZ3t27dlJ2dreDgYHl4eGjnzp1avXq1Nm/eXKLImpiYqPT0dG3atEnr168v933n/N/nV5z0Xi8hIUH169dXQECApk2bpkuXLpnatm3bpvvuu08NGzY0HQsODlZWVpYOHDhQYiw7Ozvl5uaWOH47oqOjFRoaKnt7e4WGhio6Otqsfdq0aZo3b56mT5+ugwcPasWKFaZ4U1JSJEmbN2/WyZMntXbt2hLjP/nkkzp//ry++eYb07GMjAxt2LDBNIP+WmV9j2PHjtWKFSvM7v+DDz5Q48aNFRQUZJHPAgAAAABuRVXm8JZCDl/7cvjFixfr/Pnzmj17tl544QUFBAQoIiLCrE9iYqLS0tKUlJSklStXau3atZo5c6apPTs7W6+88op27dqlxMRE2djYaPDgwSUms0+dOlUvvvii0tLSFBwcbLF7QOUrfcoHAAAAUIMtGT3sls8tKirS5mVva9P7UWrSJkDDZ8wztb0/4Vld/j2rxDmvrip/ElmWtm3basaMGZIkPz8/RUVFKTExUY899pipT3h4uEJDQyVJc+fO1ZIlS5SSkqI+ffqUOmZ+fr6ioqL0wAMPSLqauPv7+yslJUWdO3c26+vi4iJHR0fl5ubK29vbdDwuLk45OTmKj4+Xs7OzJCkqKkoDBw7Um2++aUponZ2dtWzZMjk4OJT7nvPy8rR69Wo1adLENPa1Ro4cqWbNmsnHx0d79+7VlClTlJ6ebiognzp1yiyhlmR6f+rUqRLjtWrVSp988omeeOIJ1alTp9xxliUrK0tr1qzRtm3bJElPPfWUHnroIS1evFguLi76/ffftXjxYkVFRWn06NGSpJYtW+rBBx+UJHl5eUmSPD09zT7za3l4eKhv375asWKFevXqJUlas2aN6tevr0ceeaRE/7K+xyFDhmjChAn69NNPFRISIkmKjY1VeHh4hZbuAwAAAABLq8oc3hL5u0QOXxtzeFdXVy1fvly9e/eWs7Oz9u7dWyIfdnBwUExMjJycnHTvvfdq1qxZ+stf/qLZs2fLxsZGQ4cONesfExMjLy8vHTx4UAEBAabjL730koYMGXLbMaPq8cQ4AAAA7jhFRUVVfs22bduavW/UqJHOnDlTZh9nZ2e5urqW6HMtOzs7derUyfT+nnvukbu7u9LS0sodV1pamtq1a2eW9Hbv3l2FhYVKT083HbvvvvsqlFAnJCTI0dFRH330keLj40vt89xzzyk4OFj33XefwsLCFB8fr08++cRsmbKKiI6O1ldffSUnJ6cbLmV27733ysXFRS4uLurbt2+Z/VauXKmWLVuqXbt2kqT27durWbNmWrVqlaSrn11ubq6poH2rwsLC9PHHH5tmyickJGjEiBGysSl/umY0GjVq1CjTPuh79uzR/v37FR4efluxAQAAAEB1I4f/Azl82Tm8JAUFBalLly4aNWqUmjVrVqK9Xbt2cnJyMr3v2rWrLl68aFrS/tChQwoNDdXdd98tV1dXNW/eXJJ0/Phxs3ECAwPLe5uoYXhiHAAAALXOxLg15e77e8Z5xb7yvFkibbCxUfjCd1W3fn2zvuOiYiwW4/WuX4bMYDCUWIqrPH2qS2mzxW9k0KBBSklJ0bRp0zRp0iTt3r37pucUz5o/fPiwWrZsKW9vb9Ny5MVOnz4tSaU+gT116lQFBARo0aJFN9xD7csvv1R+fr4kydHRscx+0dHROnDggNneaoWFhYqJidGYMWNueG5FDBw4UEVFRfriiy/UqVMnbdmyRW+99VaFxxk7dqzat2+vn3/+WcuXL1dQUFCpPwQAAAAAQFUih696d
```

Comment:
Is it much effort to allow users to provide a custom minimal difference?

Reply:
One could then do convergence checks

Reply:
Shouldn't be too hard. And would be nice to have, I agree. :) 
Will implement now

Reply:
Implemented as `min_diff_factor` now where the minimal difference is then `min_diff_factor` * `measurement_range` / `n_spline_pars`